tests/unit/job/test_query.py

```diff
@@ -1289,7 +1217,18 @@ def test_result_w_retry(self):
         )
 
         connection.api_request.assert_has_calls(
-            [query_results_call, query_results_call, reload_call]
+            [
+                reload_call,
```

Comment:
Same thing here. Can I get some clarity on what we are doing and looking for?

Reply:
Added some explanation here as well as above in the `make_connection()` call.